projects/geometry/110_mesh_shader_triangle_d3d12/110_mesh_shader_triangle_d3d12.cpp

```diff
@@ -0,0 +1,330 @@
+#include "window.h"
+
+#include "dx_renderer.h"
+
+#include <glm/glm.hpp>
+#include <glm/matrix.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtx/transform.hpp>
+using namespace glm;
+
+#define CHECK_CALL(FN)                               \
+    {                                                \
+        HRESULT hr = FN;                             \
+        if (FAILED(hr))                              \
+        {                                            \
+            std::stringstream ss;                    \
+            ss << "\n";                              \
+            ss << "*** FUNCTION CALL FAILED *** \n"; \
+            ss << "FUNCTION: " << #FN << "\n";       \
+            ss << "\n";                              \
+            GREX_LOG_ERROR(ss.str().c_str());        \
+            assert(false);                           \
+        }                                            \
+    }
+
+// =============================================================================
+// Shader code
+// =============================================================================
+const char* gShaders = R"(
+struct MeshOutput {
+    float4 Position : SV_POSITION;
+    float3 Color    : COLOR;
+};
+
+[outputtopology("triangle")]
+[numthreads(1, 1, 1)]
+void msmain(out indices uint3 triangles[1], out vertices MeshOutput vertices[3]) {
+    SetMeshOutputCounts(3, 1);
+    triangles[0] = uint3(0, 1, 2);
+
+    vertices[0].Position = float4(-0.5, 0.5, 0.0, 1.0);
+    vertices[0].Color = float3(1.0, 0.0, 0.0);
+
+    vertices[1].Position = float4(0.5, 0.5, 0.0, 1.0);
+    vertices[1].Color = float3(0.0, 1.0, 0.0);
+
+    vertices[2].Position = float4(0.0, -0.5, 0.0, 1.0);
+    vertices[2].Color = float3(0.0, 0.0, 1.0);
+}
+
+float4 psmain(MeshOutput input) : SV_TARGET
+{
+    return float4(input.Color, 1);
+}
+
+)";
+
+// =============================================================================
+// Globals
+// =============================================================================
+static uint32_t gWindowWidth  = 1280;
+static uint32_t gWindowHeight = 720;
+static bool     gEnableDebug  = true;
+
+static LPCWSTR gVSShaderName = L"vsmain";
+static LPCWSTR gPSShaderName = L"psmain";
+
+void CreateGlobalRootSig(DxRenderer* pRenderer, ID3D12RootSignature** ppRootSig);
+void CreateGeometryBuffers(
+    DxRenderer*      pRenderer,
+    ID3D12Resource** ppIndexBuffer,
+    ID3D12Resource** ppVertexBuffer,
+    ID3D12Resource** ppVertexColorBuffer);
+
+// =============================================================================
+// main()
+// =============================================================================
+int main(int argc, char** argv)
+{
+    std::unique_ptr<DxRenderer> renderer = std::make_unique<DxRenderer>();
+
+    if (!InitDx(renderer.get(), gEnableDebug))
+    {
+        return EXIT_FAILURE;
+    }
+
+    D3D12_FEATURE_DATA_D3D12_OPTIONS7 options7 = {};
+    //
```

Comment:
Probably don't need the spurious extra //

Reply:
Removed.